walk-in-interview/src/main/java/com/google/job/data/Job.java

```diff
@@ -1,26 +1,252 @@
 package com.google.job.data;
 
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
 /** Class for a job post. */
 public final class Job {
-    // TODO(issue/9): Add the feature to create a new job post
-    private String jobTitle;
+    private final String jobId;
+    // TODO(issue/25): merge the account stuff into job post.
+    private final JobStatus jobStatus;
+    private final String jobTitle;
+    private final Location jobLocation;
+    private final String jobDescription;
+    private final JobPayment jobPay;
+    private final List<String> requirements;
+    private final long postExpiryTimestamp;
+    private final JobDuration jobDuration;
+
+    private volatile int hashCode;
+
+    private Job(JobBuilder jobBuilder) {
+        this.jobId = jobBuilder.jobId;
+        this.jobStatus = jobBuilder.jobStatus;
+        this.jobTitle = jobBuilder.jobTitle;
+        this.jobLocation = jobBuilder.location;
+        this.jobDescription = jobBuilder.jobDescription;
+        this.jobPay = jobBuilder.jobPay;
+        this.requirements = jobBuilder.requirements;
+        this.postExpiryTimestamp = jobBuilder.postExpiryTimestamp;
+        this.jobDuration = jobBuilder.jobDuration;
+    }
 
     // No-argument constructor is needed to deserialize object when interacting with cloud firestore.
-    public Job() {}
+    public Job() {
+        this.jobId = "";
+        this.jobStatus = JobStatus.ACTIVE;
+        this.jobTitle = "";
+        this.jobLocation = new Location();
+        this.jobDescription = "";
+        this.jobPay = new JobPayment();
+        this.requirements = ImmutableList.of();
+        this.postExpiryTimestamp = 0;
+        this.jobDuration = JobDuration.OTHER;
+    }
+
+    public static JobBuilder newBuilder() {
+        return new JobBuilder();
+    }
+
+    public static final class JobBuilder {
+        // Optional parameters - initialized to default values
+        private String jobId = "";
+        private JobStatus jobStatus = JobStatus.ACTIVE;
```

Comment:
as i mentioned in another comment, i think you want job status to always be set explicitly instead of relying on default value. an "active" jobstatus means that it will be visible to the users. this should be a decision made by the person registering the job not the default value. So i would leave this as nullable and check that it is set in the build() function.